codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ServiceGenerator.java

```diff
@@ -125,13 +128,26 @@ private void generateConfig() {
             );
             writer.write("APIOptions []$L", API_OPTIONS_FUNC_NAME).write("");
 
-            for (GoIntegration integration : integrations) {
-                integration.addConfigFields(settings, model, symbolProvider, writer);
+            // Add config fields to the options struct.
+            for (ConfigField configField : getAllConfigFields()) {
+                configField.getDocumentation().ifPresent(writer::writeDocs);
+                writer.write("$L $P", configField.getName(), configField.getType());
```

Comment:
Should we generate option types as values rather then pointers? Additionally could we maybe make that configurable via the config field? `GetForatter` or something similar to determine $T vs $P

Reply:
Discussed offline, but this can be controlled by the pointable property of the symbol you're given.